### PR TITLE
tui: more work on DECSCUSR fixup logic: fix #7235

### LINKED ISSUE #7235: Reappearing of incorrect DECSCUSR codes problem
- `nvim --version`:
```
NVIM v0.2.1-826-g7bfa4406c
Build type: RelWithDebInfo
Compilation: /usr/lib64/ccache/bin/x86_64-pc-linux-gnu-gcc -O2 -march=x86-64 -pipe -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -I/var/tmp/portage/app-editors/neovim-9999/work/neovim-9999_build/config -I/var/tmp/portage/app-editors/neovim-9999/work/neovim-9999/src -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/var/tmp/portage/app-editors/neovim-9999/work/neovim-9999_build/src/nvim/auto -I/var/tmp/portage/app-editors/neovim-9999/work/neovim-9999_build/include
Compiled by root@note

Features: +acl +iconv +jemalloc +tui
See ":help feature-compile"

   system vimrc file: "/etc/vim/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"
```

- Vim (version: 8.0.0983) behaves differently? `Yes`
- Operating system/version: `Gentoo`
- Terminal name/version: `Konsole 17.08.0` + `Tmux` (issue reproduces only in tmux inside Konsole)
- `$TERM`: `tmux-256color` (changing to konsole-256color, screen.konsole-256color doesn't help)

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC
```
(happens just after start)

### Actual behaviour
symbols `q` (exact symbols depends on DECSCUSR codes) start appearing every few seconds in nvim and in the tmux window after exitting from nvim.
### Expected behaviour
Cursor shape changes works fine.

Actually it worked fine when I tested patch from #6997 some time ago. But when I rebuilt nvim after "merging" #6997 PR — it is broken again.

P.S. `KONSOLE_*` variables doesn't are kept in tmux session, so, they can't be the reason.

### PR DESCRIPTION
**Pull request by msva**

### CODE REVIEW

**Comment by msva** [comment]
ping @justinmk

File: src/nvim/tui/tui.c [hunk 1]
```diff
@@ -1448,7 +1448,27 @@ static void patch_terminfo_bugs(TUIData *data, const char *term,
...
+          "%p1%{1}%<" "%t%{1}"  // Fortunately if we exclude zero as special,
+          "%e%p1%{1}%&"  // in all other cases we can treat bit #0 as a flag.
+          "%;%d\x07"));
+      if (-1 == data->unibi_ext.reset_cursor_style) {
+          data->unibi_ext.reset_cursor_style = (int)unibi_add_ext_str(ut, "Se",
+                                                                      "");
+      }
+      unibi_set_ext_str(ut, (size_t)data->unibi_ext.reset_cursor_style,
+          "\x1b]50;\x07");
+    } else if (true_xterm    // per xterm ctlseqs doco (since version 282)
```
Comment (justinmk) [inline src/nvim/tui/tui.c]: Embedding decisions in the order of these conditions is fragile. Explicit is better. It also avoids churn.

Why wouldn't this work:

    if (!konsole && (true_xterm ...

Comment (msva) [reply src/nvim/tui/tui.c]: Ok, I've rewrite the patch with your advice. Seems to work too.

**Comment by msva** [comment]
@justinmk so, is it any ETA when it (current PR version, with `!konsole`) will be merged into master?

**Pull request merged**